File: frontend/src/features/trainers/types/trainer.ts
// 백엔드 TrainerResponse 와 1:1 대응 (domain/trainer/dto/TrainerResponse)
export interface Trainer {
  id: number;
  name: string;
  phone: string | null;
  // 퇴사 등으로 비활성 처리된 트레이너. 지난 예약이 참조하므로 지우지 않는다 — 새 예약만 막힌다.
  active: boolean;
}

// 등록·수정 공용 요청. active 는 수정에서만 의미가 있고, 빼면 서버가 그대로 둔다.
export interface TrainerSaveRequest {
  name: string;
  phone: string | null;
  active?: boolean;
}

File: backend/src/main/java/com/gym/domain/trainer/entity/Trainer.java
package com.gym.domain.trainer.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.Instant;
import org.hibernate.annotations.CreationTimestamp;

/**
 * 트레이너 (docs/ERD.md §3.7). PT 예약 기록이 참조하므로 지우지 않고 active 로 비활성 처리한다.
 * 비활성 트레이너에게는 새 예약을 넣을 수 없지만, 지난 예약 기록은 그대로 보인다.
 */
@Entity
@Table(name = "trainer")
public class Trainer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private String phone;

    @Column(nullable = false)
    private boolean active;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    protected Trainer() {
        // JPA 전용 기본 생성자
    }

    public Trainer(String name, String phone) {
        this.name = name;
        this.phone = phone;
        this.active = true;
    }

    public void updateInfo(String name, String phone) {
        this.name = name;
        this.phone = phone;
    }

    public void changeActive(boolean active) {
        this.active = active;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getPhone() {
        return phone;
    }

    public boolean isActive() {
        return active;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }
}
